cmd/nerdctl/container_run.go

```diff
@@ -724,12 +733,36 @@ func createContainer(ctx context.Context, cmd *cobra.Command, client *containerd
 
 	var s specs.Spec
 	spec := containerd.WithSpec(&s, opts...)
+
 	cOpts = append(cOpts, spec)
 
-	container, err := client.NewContainer(ctx, id, cOpts...)
-	if err != nil {
+	container, cerr := client.NewContainer(ctx, id, cOpts...)
+	var netErr error
+	// NOTE: on non-Windows platforms, network setup is performed by OCI hooks.
+	// Seeing as though Windows does not currently support OCI hooks, we must explicitly
+	// perform the network cleanup in the main nerdctl executable.
+	if runtime.GOOS == "windows" {
+		netErr = netManager.SetupNetworking(ctx, id)
+	}
+
+	if cerr != nil || netErr != nil {
+		err := cerr
+		if err == nil {
+			err = netErr
+		}
+
 		gcContainer := func() {
 			var isErr bool
+			if netErr != nil {
+				logrus.Warnf("a networking setup error has occurred, reverting networking setup: %s", netErr)
+
+				errE := netManager.CleanupNetworking(ctx, container)
```

Comment:
should this only be called for Windows as it is above?

Reply:
Correct, and considering `netErr != nil` was only a possibility on Windows this was already a given, but I'll try tomake it more explicit.